projects/distributed-tracing/src/shared/dashboard/widgets/table/table-widget-select-filter.model.ts

```diff
@@ -0,0 +1,42 @@
+import { Dictionary, PrimitiveValue } from '@hypertrace/common';
+import { FilterOperator, TableFilter } from '@hypertrace/components';
+import { Model, ModelApi, ModelProperty, STRING_PROPERTY } from '@hypertrace/hyperdash';
+import { ModelInject, MODEL_API } from '@hypertrace/hyperdash-angular';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Model({
+  type: 'table-widget-select-filter',
+  displayName: 'Select Filter'
+})
+export class TableWidgetSelectFilterModel {
+  @ModelProperty({
+    key: 'attribute',
+    type: STRING_PROPERTY.type,
+    required: true
+  })
+  public attribute!: string;
+
+  @ModelProperty({
+    key: 'placeholder',
+    type: STRING_PROPERTY.type
+  })
+  public placeholder?: string;
+
+  @ModelInject(MODEL_API)
+  protected readonly api!: ModelApi;
+
+  public getData(): Observable<PrimitiveValue[]> {
+    return this.api
+      .getData<Dictionary<PrimitiveValue>[]>()
```

Comment:
I proposed one tweak in another comment, but alternately, do we even need a fetch? Could we use the fetched data to populate this and just pass the data to the filter model to transform into a filters? That's basically what we're doing with the existing table filters right? The downside would be the same, that our filter options are only based on the queried data.

Reply:
Aaaaah, right. Agree they should be consistent. Fixed mine to just provide an array of attributes instead of the full Entity. Much better.

As for the single fetch, I think this the independent fetch is still a better route. I don't really like the way I did the header filters where we tie it to the table data. It couple everything too closely. Granted, if you are filtering on a table, the data between the filer and the table should match so technically they _are_ coupled. 

But by having them independently fetched we have more freedom to fetch results that might not make be fetched by the table due to pagination or other filters (maybe debatable if that second part is good or bad). It also allows us to have filters for columns we don't even show.

I think I'd like to go with what we have now (along with your other suggestions).